pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3183,6 +3186,20 @@ public String startReplaceSegments(String tableNameWithType, List<String> segmen
     return segmentLineageEntryId;
   }
 
+  private void waitForSegmentsToDelete(String tableNameWithType, List<String> segments, long timeOutInMillis)
+      throws InterruptedException {
+    LOGGER.info("Wait segments to delete. timeout = {}ms, segments = {}", timeOutInMillis, segments);
+    long endTimeMs = System.currentTimeMillis() + timeOutInMillis;
+    do {
+      if (Collections.disjoint(getSegmentsFor(tableNameWithType, false), segments)) {
+        return;
+      } else {
+        Thread.sleep(SEGMENT_CLEANUP_CHECK_INTERVAL_MS);
+      }
+    } while (System.currentTimeMillis() < endTimeMs);
+    throw new RuntimeException("Timeout while waiting for segments to be deleted");
```

Comment:
log the table name and the timeout value here.

Reply:
Also, suggest adding some details of the exception in Line 3176 to briefly explain why the request failed.

Reply:
logged the table name and timeout.
We are logging the exception on `L3178`, so we don't need any other details?